projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt

```diff
@@ -199,11 +233,34 @@ class CommunityApiController(
                 {
                     communityApiService.compare(
                         Compare(params = mapOf("crn" to crn) + params, uri = it.key),
-                        headers
+                        headers,
+                        showValues = true
                     )
                 }, taskExecutor
             )
         }
     }
+
+    fun sendComparisonReport(params: Map<String, Any>, uri: Uri, request: HttpServletRequest) {
+        CompletableFuture.supplyAsync({
+            val headers = mapOf(HttpHeaders.AUTHORIZATION to request.getHeader(HttpHeaders.AUTHORIZATION))
+            val report = communityApiService.compare(Compare(params = params, uri = uri.name), headers)
+            telemetryService.comparisonFailureEvent(params["crn"].toString(), report)
+        }, taskExecutor)
+    }
+
+    fun TelemetryService.comparisonFailureEvent(crn: String, compareReport: CompareReport) {
+        if (!compareReport.success) {
+            val issuesMap = compareReport.issues?.mapIndexed { index, s -> index.toString() to s }?.toMap()!!
+            trackEvent(
+                "ComparisonFailureEvent",
+                mapOf("crn" to crn) +
+                    mapOf("endpointName" to compareReport.endPointName) +
+                    mapOf("url" to compareReport.url!!) +
+                    mapOf("numberOfDifferences" to compareReport.issues.size.toString()) +
+                    issuesMap
```

Comment:
You should be able to use a single `mapOf` here.  Also could we use a static key for the "differences" e.g.
```suggestion
                mapOf(
                    "crn" to crn,
                    "endpointName" to compareReport.endPointName,
                    "url" to compareReport.url!!,
                    "numberOfDifferences" to compareReport.issues.size.toString(),
                    "differences" to compareReport.issues.toString(),
                )
```